platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/DefaultProblemsReportCreator.kt

```diff
@@ -0,0 +1,125 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.problems.internal
+
+import org.gradle.api.internal.DocumentationRegistry
+import org.gradle.api.internal.StartParameterInternal
+import org.gradle.api.internal.file.temp.TemporaryFileProvider
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.problems.internal.FileLocation
+import org.gradle.api.problems.internal.Problem
+import org.gradle.internal.buildoption.InternalOptions
+import org.gradle.internal.cc.impl.problems.BuildNameProvider
+import org.gradle.internal.cc.impl.problems.JsonSource
+import org.gradle.internal.cc.impl.problems.JsonWriter
+import org.gradle.internal.concurrent.ExecutorFactory
+import org.gradle.internal.configuration.problems.CommonReport
+import org.gradle.internal.configuration.problems.FailureDecorator
+import org.gradle.internal.configuration.problems.StructuredMessage
+import org.gradle.internal.configuration.problems.writeError
+import org.gradle.internal.configuration.problems.writeStructuredMessage
+import org.gradle.internal.logging.ConsoleRenderer
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.problems.failure.FailureFactory
+import org.gradle.problems.buildtree.ProblemReporter
+import java.io.File
+
+val logger: Logger = Logging.getLogger(DefaultProblemsReportCreator::class.java)
+
+class DefaultProblemsReportCreator(
+    executorFactory: ExecutorFactory?,
+    temporaryFileProvider: TemporaryFileProvider?,
+    internalOptions: InternalOptions?,
+    startParameter: StartParameterInternal,
+    private val failureFactory: FailureFactory,
+    private val buildNameProvider: BuildNameProvider
+) : ProblemReportCreator {
+
+    private val report = CommonReport(executorFactory!!, temporaryFileProvider!!, internalOptions!!, "problem-report")
+    private val taskNames: List<String> = startParameter.taskNames
+    private var problemCount = 0
+
+    private val failureDecorator = FailureDecorator()
+
+    override fun getId(): String {
+        return "DefaultProblemsReportCreator"
+    }
+
+    override fun report(reportDir: File, validationFailures: ProblemReporter.ProblemConsumer) {
+        report.writeReportFileTo(reportDir, object : JsonSource {
+            override fun writeToJson(jsonWriter: JsonWriter) {
+                with(jsonWriter) {
+                    property("reportContext", "problems report")
+                    property("totalProblemCount", problemCount.toString())
+                    buildNameProvider.buildName()?.let { property("buildName", it) }
+                    property("requestedTasks", taskNames.joinToString(" "))
+                    property("cacheAction", "cacheAction")
+                    property("cacheActionDescription") {
+                        writeStructuredMessage(StructuredMessage.Builder().text("").build())
+                    }
```

Comment:
❌  There should be no need in writing CC-specific data to the model.

If this is done to accommodate the requirements on the data conversion on the report side, I suggest we merge the changes required on the report side first, before proceeding with the PR on the `gradle/gradle` side.